python/graphstorm/eval/evaluator.py

```diff
@@ -370,26 +377,31 @@ class GSgnnRegressionEvaluator(GSgnnInstanceEvaluator):
         1) consecutive_increase and 2) average_increase.
     """
     def __init__(self, eval_frequency,
-                 eval_metric,
+                 eval_metric=None,
+                 multilabel=False,
                  use_early_stop=False,
                  early_stop_burnin_rounds=0,
                  early_stop_rounds=3,
-                 early_stop_strategy=EARLY_STOP_AVERAGE_INCREASE_STRATEGY):
-        super(GSgnnRegressionEvaluator, self).__init__(eval_frequency,
+                 early_stop_strategy=EARLY_STOP_AVERAGE_INCREASE_STRATEGY): # pylint: disable=unused-argument
+        # set up default metric to be accuracy
+        if eval_metric is None:
+            eval_metric = ["accuracy"]
+        super(GSgnnClassificationEvaluator, self).__init__(eval_frequency,
             eval_metric, use_early_stop, early_stop_burnin_rounds,
             early_stop_rounds, early_stop_strategy)
+        self._multilabel = multilabel
         self._best_val_score = {}
         self._best_test_score = {}
         self._best_iter = {}
-        self.metrics_obj = RegressionMetrics()
+        self.metrics_obj = ClassificationMetrics(multilabel=self._multilabel)
 
         for metric in self.metric:
             self.metrics_obj.assert_supported_metric(metric=metric)
             self._best_val_score[metric] = self.metrics_obj.init_best_metric(metric=metric)
             self._best_test_score[metric] = self.metrics_obj.init_best_metric(metric=metric)
             self._best_iter[metric] = 0
 
-    def evaluate(self, val_pred, test_pred, val_labels, test_labels, total_iters):
+    def evaluate(self, val_pred, test_pred, val_labels, test_labels, total_iters): # pylint: disable=arguments-differ
```

Comment:
This is not a good exercise to change the argument lists of the functions of the parent class. 
It is better to define a GSgnnPredictionEvalInterface to define the eval function for classification/regression evaluators and GSgnnRankingEvalInterface for lp evaluators. Similar to GSgnnNodeModelInterface, GSgnnEdgeModelInterface.

Reply:
Thanks. Seperate the eval functions in Interface classes. Will use this method in the next PRs.